mockkubeapiserver/storage/memorystorage/memorystorage_watch.go

```diff
@@ -0,0 +1,119 @@
+package memorystorage
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/kubebuilder-declarative-pattern/mockkubeapiserver/storage"
+)
+
+func (resource *memoryResourceInfo) Watch(ctx context.Context, opt storage.WatchOptions, callback storage.WatchCallback) error {
+	return resource.storage.watch(ctx, resource.gvk, opt, callback)
+}
+
+type resourceStorage struct {
+	parent        *MemoryStorage
+	GroupResource schema.GroupResource
+
+	mutex   sync.Mutex
+	watches []*watch
+
+	resourceVersionClock *resourceVersionClock
+
+	objects map[types.NamespacedName]*unstructured.Unstructured
+}
+
+type resourceVersionClock struct {
+	now atomic.Int64
```

Comment:
I missed the atomic in the previous review. This is cool.

Reply:
Yes, in general it's premature optimization vs sync.Mutex, but here I find it more ergonomic (as well as being possibly more efficient)